python/lsst/daf/butler/tests/_testRepo.py

```
@@ -68,13 +72,16 @@ def makeTestRepo(root, dataIds, *, config=None, **kwargs):
     `expandUniqueId`, so long as no other code has inserted dimensions into
     the repository registry.
     """
-    if not config:
-        config = Config()
-        config["datastore", "cls"] = "lsst.daf.butler.datastores.inMemoryDatastore.InMemoryDatastore"
-        config["datastore", "checksum"] = False  # In case of future changes
-        config["registry", "db"] = "sqlite:///:memory:"
+    defaults = Config()
+    defaults["datastore", "cls"] = "lsst.daf.butler.datastores.inMemoryDatastore.InMemoryDatastore"
+    defaults["datastore", "checksum"] = False  # In case of future changes
+    defaults["registry", "db"] = "sqlite:///:memory:"
+
+    if config:
+        defaults.update(config)
+
     # newConfig guards against location-related keywords like outfile
-    newConfig = Butler.makeRepo(root, config=config, **kwargs)
+    newConfig = Butler.makeRepo(root, config=defaults, **kwargs)
```

I like this approach, but the name `defaults` is misleading once you call `update`. Maybe `mergedConfig` to distinguish it from the function parameter?